includes/razorpay-subscriptions.php

```diff
@@ -124,10 +132,86 @@ protected function getSubscriptionCreateData($orderId)
         return $subscriptionData;
     }
 
-    protected function getProductPlanId($product, $order)
+    /**
+     * @param $subscriptionData
+     * @param $signUpFee
+     * @param $order
+     * @throws Errors\Error
+     */
+    protected function setStartAtIfNeeded(& $subscriptionData, & $signUpFee, $order)
     {
-        $currency = get_woocommerce_currency();
+        $product = $this->getProductFromOrder($order);
+
+        $sub = $this->getWooCommerceSubscriptionFromOrderId($order->get_id());
+
+        $metadata = get_post_meta($product['product_id']);
+
+        if (empty($metadata['razorpay_wc_start_date']) === false)
+        {
+            //
+            // When custom start date is set, we consider the initial payment
+            // as a sign up payment, and the first recurring payment will be
+            // made on the configured start date on the next month
+            //
+            $startDay = $metadata['razorpay_wc_start_date'][0];
+
+            $startDay = 40;
+
+            //
+            // $startDay must be in between 1 and 28
+            //
+            if (($startDay <= 1) or ($startDay >= 28))
+            {
+                throw new Errors\Error(
+                    'Invalid start day saved as subscription product metadata',
+                    WooErrors\SubscriptionErrorCode::SUBSCRIPTION_START_DATE_INVALID,
+                    400
+                );
+            }
+
+            $startDate = $this->getStartDate($startDay, $sub);
+
+            // We modify the sign up fee which was passed by reference
+            $signUpFee += $sub->get_total();
+
+            $subscriptionData['start_at'] = $startDate;
+
+            //
+            // In the case where we take the first recurring payment as a up front amount, and the
+            // second recurring payment as the first recurring payment, we reduce the total count by 1
+            //
+            $subscriptionData['total_count'] = $subscriptionData['total_count'] - 1;
+        }
+    }
 
+    protected function getStartDate($startDay, $sub)
+    {
+        $period = $sub->get_billing_period();
+
+        $interval = $sub->get_billing_interval();
+
+        $date = new DateTime('now', new DateTimeZone('Asia/Kolkata'));
```

Comment:
Use the default timezone? Why use IST?

Reply:
@captn3m0 Well that would mean we would process the payments based on UTC timestamps. But wouldn't that cause some issues?

Reply:
@captn3m0 Well that would mean we would process the payments based on UTC timestamps. But wouldn't that cause some issues? 

For eg. Oct 21st 2:30 IST would convert to Nov 3rd 10pm based on this code. Using IST would mean Oct 21st 2:30 would become Nov 21st 2:30 -> Nov 3rd 2:30 finally.

Reply:
The start date is ultimately a unix timestamp, which is what matters. If someone selects 28, and the website is in +4:30 instead of +5:30, we should respect that. Moreover, we should clarify this somewhere (how to check your timezone, in the doc for this feature specifically).

Reply:
@captn3m0 Yeah, but the default timezone is always `UTC`. One would have to manually initialize the `DateTime` object with the current timezone. So either we default to `IST` or leave it at `UTC`

Reply:
@captn3m0 As discussed, using `UTC`.